public/service-worker.js

```diff
@@ -227,9 +227,16 @@ self.addEventListener("install", (event) => {
 self.addEventListener("activate", (event) => {
   event.waitUntil(
     (async () => {
+      // Optimizes startup time for the SW with preloading responses
+      // https://web.dev/blog/navigation-preload
+      // see also the implementeation of handleNavigationRequest
       if ("navigationPreload" in self.registration) {
         await self.registration.navigationPreload.enable();
       }
+
+      // iterate over all the cache keys and delete everything but
```

Comment:
This function removes caches that are no longer up to date. If a service worker is updated, it may no longer need all of the data in some of the caches. When we delete these cache before creating new ones, we avoid creating bloat data and make sure that we always use the intended data. 

With the current solution two separate caches are used (`OFFLINE_CACHE ` and `DATA_CACHE`). A third cache (`NOTIFICATION_CACHE`) was used for storing timestamps, for the user's last received notification and the most recent engagement with the app, but isn't used right now in the current solution.